### DISCUSSION TITLE
Weird obtained results from SSL pre-training

### DISCUSSION THREAD

**Discussion by ahkarami** [NVIDIA-NeMo/NeMo#3896] [Q&A] [answered]
@titu1994 
@sam1373 
Hi,
I have used the SSL pre-training of NeMo script via CitriNet512 encoder (with the mentioned config of [this link](https://github.com/NVIDIA/NeMo/issues/3792)) on ~ a 460 Hours **persian** data set. The raw audio files has been segmented to 15 seconds uniform-length utterances for the SSL pre-training data set. I have trained it for 100 epochs and it seems to converge nicely (its contrastive loss decreased from ~10,000 to 6,895). After that, I have fine-tuned the pre-trained model on a ~10 Hours labeled data set and I got the `46.14%` WER on my test set (without any LM). 
At the second experiment, I just loaded the pre-trained English CitriNet512 model weight and fine-tune it on the mentioned 10 Hours labeled data set with the exactly same training parameters of the previous training experiment. In this experiment, I have got the `36.37%` WER on my test set (without any LM)!! 
This obtained results are very strange for me, because in the first experiment I have used the pre-trained SSL model which have trained on **Persian** data set (with SSL strategy), and I expected that its obtained result must be better than the English pre-trained model. However, the NeMo English pre-trained model (without any SSL pre-training) achieved ~ `10%` better WER!! 
At the third experiment, I have repeated the above experiment without any pre-trained weight, and I got the `96.88%` WER on the mentioned test set (without any LM). The below table summarized the above-mentioned results:
| **Experiment on 10H Labeled Persian Data Set**  | **WER** |
| --------------- | --------------- |
| **SSL pre-trained** | `46.14%` |
| **English pre-trained** | `36.37%` |
| **without pre-trained** | `96.88%` |

Do you think I made a mistake somewhere (_e.g._, SSL pre-training configs) or is it natural that the result of the English pre-trained model is better than the SSL pre-trained model on a non-English language? 
Best

**Comment by sam1373** [comment, marked as answer]
Hi @ahkarami,
This result is possible even without any mistakes, as you are comparing a model pre-trained on only 460 hours of unlabeled data to one pre-trained on a much larger amount of clean labeled data. Generally to get good results from SSL you need a much larger amount of hours, so in this case it is not surprising that having the data be of the same language is not enough of an advantage for the ssl model.

**Reply by ahkarami** [reply]
Thanks for your answer. 
Best

**Reply by titu1994** [reply]
Sidenote, the pretrained model was trained on roughly 7000 hours of labeled data, so the order of magnitude of labeled clean training data is a major factor in play here.
